Source/CarthageKit/Project.swift

```diff
@@ -728,6 +756,47 @@ private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL:
 		|> take(1)
 }
 
+/// Sends the URL to each bcsymbolmap found in the given directory.
+private func BCSymbolMapsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL, fileExtension: "bcsymbolmap")
+}
+
+/// Sends the URLs of the bcsymbolmap files that match the given framework and are
+/// located somewhere within the given directory.
+private func BCSymbolMapsForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return UUIDsForFramework(frameworkURL)
+		|> flatMap(.Merge) { UUIDs in
+			if UUIDs.isEmpty {
+				return SignalProducer.empty
+			}
+			// conceptually, we just want to filter the URLs to find any that are in uuids.
+			// but we'd like to stop enumeration early when we find all the files.
+			// this also guarantees we don't send two URLs for bcsymbolmap files with the same UUID.
+			return SignalProducer { observer, disposable in
+				var remainingUUIDs = UUIDs
+				BCSymbolMapsInDirectory(directoryURL)
+					|> startWithSignal { signal, signalDisposable in
+						disposable += signalDisposable
+						signal.observe(Signal.Observer { event in
+							switch event {
+							case let .Next(value):
+								if let basename = value.value.lastPathComponent?.stringByDeletingPathExtension, fileUUID = NSUUID(UUIDString: basename) {
+									if remainingUUIDs.remove(fileUUID) != nil {
+										observer.put(event)
+										if remainingUUIDs.isEmpty {
+											sendCompleted(observer)
```

Comment:
All that said, I think that I can actually wrap the call to `filter()` in a closure that creates the dictionary there. It's a little confusing because `filter()` is actually a transform on `Signal`s that's being lifted to a `SignalProducer`, which is why I didn't think of it before. I'll see if I can whip that up.